services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -594,15 +578,84 @@ def _compute_statistics(
         )
 
     def compute_and_prepare_response(self, data: pl.DataFrame) -> StatisticsPerColumnItem:
-        stats = self._compute_statistics(data, self.name, self.n_samples, n_bins=self.n_bins)
+        stats = self._compute_statistics(data, column_name=self.name, n_samples=self.n_samples, n_bins=self.n_bins)
         return StatisticsPerColumnItem(
             column_name=self.name,
             column_type=ColumnType.LIST,
             column_statistics=stats,
         )
 
 
-SupportedColumns = Union[ClassLabelColumn, IntColumn, FloatColumn, StringColumn, BoolColumn, ListColumn]
+class AudioColumn(Column):
+    def __init__(self, *args: Any, n_bins: int, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.n_bins = n_bins
+
+    @staticmethod
+    def get_duration(example: dict[str, Any]) -> float:
+        with io.BytesIO(example["bytes"]) as f:
+            return librosa.get_duration(path=f)  # type: ignore   # expects PathLike but BytesIO also works
+
+    @staticmethod
+    @raise_with_column_name
+    def _compute_statistics(
+        parquet_directory: Path,
+        column_name: str,
+        n_samples: int,
+        n_bins: int,
+    ) -> NumericalStatisticsItem:
+        logging.info(f"Compute statistics for Audio column {column_name} with librosa and polars. ")
+        parquet_files = list(parquet_directory.glob("*.parquet"))
+        durations = []
+        for filename in parquet_files:
+            shard_audios = pq.read_table(filename, columns=[column_name]).drop_null().to_pydict()[column_name]
```

Comment:
This can lead to quite some memory usage. For now it's maybe fine but it fails too often you can iterate on row groups instead of loading the full table.

Reply:
good idea. As even the row groups can be big

Reply:
> This can lead to quite some memory usage.

I think it would take up to ~1Gb peak usage at each iteration because max parquet file size is ~500Mb and when it contains audio bytes, the size is more or less equal to the uncompressed size. But maybe if audio files are big, loading with librosa in parallel might increase memory usage significantly, I just haven't encountered this case because usually audio datasets contain short segments of texts.  So yeah, we will see